src/shared/openssl-util.c

```diff
@@ -109,6 +110,80 @@ int rsa_pkey_to_suitable_key_size(
         return 0;
 }
 
+static int pkey_generate_volume_key_rsa(
+                EVP_PKEY *pkey,
+                void **ret_decrypted_key,
+                size_t *ret_decrypted_key_size,
+                void **ret_savedata,
+                size_t *ret_savedata_size) {
+
+        assert(pkey);
+        assert(ret_decrypted_key);
+        assert(ret_savedata);
+        assert(ret_savedata_size);
+
+        int r = rsa_pkey_to_suitable_key_size(pkey, ret_decrypted_key_size);
+        if (r < 0)
+                return log_error_errno(r, "Failed to determine RSA public key size.");
+
+        log_debug("Generating %zu bytes random key.", *ret_decrypted_key_size);
+
+        *ret_decrypted_key = malloc(*ret_decrypted_key_size);
```

Comment:
in systemd we have the rule that output parameters should never be touched on failure, and always initialized on success. Here you touch the return parameter even though the function can still fail.

hence, please write the result to a local temporary variable first, and then move it over to `*ret_decrypted_key` only once the function cannot fail anymore